feat: add new use cases section (#86)
PrestaShopCorp/docs.cloud.prestashop.com: docs/src/5-prestashop-billing/5-use-cases/README.md

> ---

Reviewer: **suggestion**: j'aurais plutôt fait 1 page par use case ça permet à la personne qui li la doc de voir directement les uses case existants

Author: Comme les How-Tos ?

title: Use Cases
---

[[toc]]

# Concepts

Prestashop Billing currently supports four types of pricing models:

- Flat fee
- Stairstep
- Usage-based
- Charge at term

Each pricing model has its advantages and suitability depending on the nature of the SaaS product, target audience, and the specific needs of the customers

> ### Flat Fee

Reviewer: **suggestion**: indiquer avec un `:::tip` que pour faire du flat fee il suffit de suivre le tutorial pour une version minimaliste et qu'ils peuvent ajouter certains des comportements de la partie How Tos

In the flat fee pricing model, customers are charged a fixed and consistent amount for using the SaaS product or service. Regardless of the usage volume or frequency, the cost remains the same. This model offers simplicity and predictability, making it easy for customers to understand and budget their expenses. It is particularly suitable for users who require unlimited access to the SaaS product without worrying about usage-based charges.

> **_Example:_** A cloud storage SaaS offers a flat fee pricing model of 10€ per month for 100GB of storage space. Regardless of whether the customer uses 50GB, 100GB, or even less than 100GB, the monthly cost remains fixed at 10€.

Reviewer: **todo**: pourrais-tu stp remplacer cet exemple par celui là 

```
Example: A mass data management SaaS offers a flat fee pricing model of 30€ per month for a basic plan and 60€ for a plan with more features.
```

Après review de la page notion https://www.notion.so/prestashopcorp/What-is-PS-Billing-90e9f1357ec84fde8ec908f21b83eb59 on préfère un exemple qui ne parle pas de quantité par peur que ce soit confus

Author: Oui pas de soucis, j'ai repris ce qu'il y avait dans le ticket ;) 

Effectivement ça sera plus parlant comme ça :)

> ### Stairstep

Reviewer: **suggestion**: indiquer avec un `:::tip` que pour faire du stairstep il faut faire le tuto + le how tos stairstep http://localhost:8081/5-prestashop-billing/4-how-tos/1-stair-step/

### Usage-Based

In the usage-based pricing model, customers are charged based on their actual usage of the SaaS product or service. The pricing is directly linked to specific usage metrics, such as the number of active users, data storage utilized, API calls made, or other measurable parameters. Customers pay only for the resources they consume, making it a flexible option for businesses with varying usage needs. This model is ideal for SaaS products with fluctuating or unpredictable usage patterns.

**_Example:_**
A video conferencing SaaS applies usage-based pricing for its business customers. They charge $0.05 per minute of video conferencing usage. If a customer conducts a 30-minute video conference, the cost would be 1.50€ (30 minutes \* 0.05€ per minute) invoiced at the end of the billing period.

:::tip

> For more details on Usage-Based API is [available here](https://prestashop-billing.stoplight.io/docs/api-gateway/896ba54736b77-add-an-usage-to-a-subscription-item) (opens new window).

Reviewer: ```suggestion
More details about Usage-Based API are [available here](https://prestashop-billing.stoplight.io/docs/api-gateway/896ba54736b77-add-an-usage-to-a-subscription-item) (opens new window).
```

Reviewer: **suggestion**: on pourrait indiquer que la version usage based peut etre cumulé avec du flat fee

:::

> ### Charge at Term

Reviewer: **suggestion**: on pourrait indiquer que la version charge at terme peut etre cumulée avec du flat fee

The charge at term pricing model involves billing customers at fixed intervals, such as monthly or annually, regardless of their actual usage during that period. The pricing is usually based on a predefined set of features or service tiers offered by the SaaS provider. This model is suitable for SaaS that have many different prices such as parcel delivery services (which have variable prices depending on the weight, size and shipping address of the parcels)

**_Example:_**
A shipping company applies charge at term pricing for its business customer. They got a lot of different pricing regarding the size or the weight of each package. If a customer send 4 packages during the month with different pricing (2 at 12€, 1 at 5$ and 1 at 35€), the cost would be 64€ invoiced at the end of the billing period.

:::tip

> For more details on Charge at Term API is [available here](https://prestashop-billing.stoplight.io/docs/api-gateway/0dec9ae31dd59-add-a-charge-at-term) (opens new window).

Reviewer: ```suggestion
More details about Charge at Term API are [available here](https://prestashop-billing.stoplight.io/docs/api-gateway/0dec9ae31dd59-add-a-charge-at-term) (opens new window).
```
